src/utils/s3_mods.py

```diff
@@ -167,3 +174,341 @@ def rd_write_feather(filepath, df):
 def rd_read_feather(filepath):
     """Placeholder Function to read feather file from HDFS"""
     return None
+
+
+def rd_file_size(filepath: str) -> int:
+    """Function to check the size of a file on s3 bucket.
+
+    Args:
+        filepath (string) -- The filepath in s3 bucket
+
+    Returns:
+        Int - an integer value indicating the size
+        of the file in bytes
+    """
+
+    _response = s3_client.head_object(Bucket=s3_bucket, Key=filepath)
+    file_size = _response['ContentLength']
+
+    return file_size
+
+
+def rd_delete_file(filepath: str) -> bool:
+    """
+    Delete a file from s3 bucket.
+    Args:
+        filepath (string): The filepath in s3 bucket to be deleted
+    Returns:
+        status (bool): True for successfully completed deletion. Else False.
+    """
+    status = delete_file(s3_client, s3_bucket, filepath)
+    return status
+
+
+def rd_md5sum(filepath: str) -> str:
+    """
+    Get md5sum of a specific file on s3.
+    Args:
+        filepath (string): The filepath in s3 bucket.
+    Returns:
+        md5result (int): The control sum md5.
+    """
+
+    try:
+        md5result = s3_client.head_object(
+            Bucket=s3_bucket,
+            Key=filepath
+        )['ETag'][1:-1]
+    except s3_client.exceptions.ClientError as e:
+        s3_logger.error(f"Failed to compute the md5 checksum: {str(e)}")
+        md5result = None
+    return md5result
+
+
+def rd_isdir(dirpath: str) -> bool:
+    """
+    Test if directory exists in s3 bucket.
+
+    Args:
+        dirpath (string): The "directory" path in s3 bucket.
+    Returns:
+        status (bool): True if the dirpath is a directory, false otherwise.
+
+    """
+    # The directory name must end with forward slash
+    if not dirpath.endswith('/'):
+        dirpath = dirpath + '/'
+
+    # Any slashes at the beginning should be removed
+    while dirpath.startswith('/'):
+        dirpath = dirpath[1:]
+
+    # Use the function from rdsa_utils
+    response = is_s3_directory(
+        client=s3_client,
+        bucket_name=s3_bucket,
+        object_name=dirpath
+    )
+    return response
```

Comment:
For the future- this doesn't fail nicely if it isn't a directory, so how about using try / except?